packages/legend-query-builder/src/stores/__tests__/QueryBuilderBuildLambdaRoundtrip.test.tsx

```diff
@@ -273,6 +281,19 @@ const cases: RoundtripTestCase[] = [
     TEST_DATA__lambda_setOperatorFilter,
     undefined,
   ],
+  // watermark
+  [
+    'Watermark with constant',
+    forWatermarkCtx,
+    TEST_DATA_lambda_watermark_Constant,
+    undefined,
+  ],
+  [
+    'Watermark with parameter',
+    forWatermarkCtx,
+    TEST_DATA_lambda_watermark_Parameter,
+    undefined,
+  ],
   [
```

Comment:
lets add a test for having query logic after the waterMark. 
so forWatermark()->project() , forWatermark()->filter().